File: 6. Python/EX38.py
# Função para ler e validar a quantidade de casos
def entDados():

    n = 0           # Inicializa n
    while n <= 0:   # Loop até receber um número positivo
        try:
            n = int(input("# Quantos casos voce vai digitar: "))  # Lê a quantidade de casos

        except Exception:
            print("# ERRO !")

    return n  # Retorna a quantidade válida

# Função para realizar a análise da divisão
def analise(num, den):

    # Realiza a divisão e exibe o resultado ou mensagem de erro
    print(f"* DIVISAO = {(num/den):.2f}\n") if den != 0 else print("* DIVISAO IMPOSSIVEL\n")

# Função principal
def main():

    # Loop para processar cada caso
    for i in range(entDados()):

        # Lê numerador e denominador e chama a função de análise
        analise(float(input("# Entre com o numerador: ")),
                float(input("# Entre com o denominador: ")))

# Verifica se o script está sendo executado diretamente
if __name__ == '__main__':
    main()  # Chama a função principal

File: 4. Java/EX38.java
package Apostila;

import java.util.Locale;
import java.util.Scanner;

public class EX38
{
    public static void main(String[] args)
    {
        // Variáveis
        int n = 0;
        double num, den;
        
        // Define o locale para formato de números
        Locale.setDefault(Locale.US);
        
        // Lê a quantidade de casos até que seja maior que zero
        try (Scanner ent = new Scanner(System.in))
        {
            while (n <= 0)
            {
                System.out.print("# Quantos casos voce vai digitar: ");
                n = ent.nextInt();
            }
            
            // Laço para realizar divisões
            for (int i = 0; i < n; i++)
            {
                // Lê o numerador
                System.out.print("# Entre com o numerador: ");
                num = ent.nextDouble();
                
                // Lê o denominador
                System.out.print("# Entre com o denominador: ");
                den = ent.nextDouble();
                
                // Verifica se o denominador é diferente de zero
                if (den != 0)
                {
                    System.out.format("* DIVISAO = %.2f \n\n", (num / den));
                }
                
                else
                {
                    // Mensagem de erro para divisão por zero
                    System.out.println("* DIVISAO IMPOSSIVEL \n");
                }
            }
        }
    }
}
